### ISSUE TITLE
Textual Item import fails if there are multiple links

### ISSUE THREAD

**Issue by kaikreuzer** [openhab/openhab-webui#878]
## The problem

I am trying to use "Add Items from Textual Definition".
My item definition has multiple links:
```
Dimmer Light_Ceiling "Deckenlicht" <light> { channel="hue:0220:1:20:brightness,knx:device:ip:knx_1_1_1:3" }
```
UI shows the message:

![Screenshot 2021-02-05 at 11 15 25](https://user-images.githubusercontent.com/3244965/107020717-7e3a1d00-67a3-11eb-9620-e8ae69263657.png)

## Expected behavior

Item and two links are created.

## Steps to reproduce

1. Add above item example to the import in the main UI.

## Your environment

Distro build 2182

## Browser console

Shows HTTP 500 server error

## Browser network traffic

HTTP 500 is returned by the call to `/rest/links/Light_Ceiling/hue%3A0220%1%3A20%3Abrightness%2Cknx%3Adevice%3Aip%3Aknx_1_1_1%3A3` which results in an exception on server side:

```json
"{\"error\":{\"message\":\"ID segment \\u00271b,knx\\u0027 contains invalid characters. Each segment of the ID must match the pattern [\\\\w-]*.\",\"http-code\":500,\"exception\":{\"class\":\"java.lang.IllegalArgumentException\",\"message\":\"ID segment \\u00271b,knx\\u0027 contains invalid characters. Each segment of the ID must match the pattern [\\\\w-]*.\",\"localized-message\":\"ID segment \\u00271b,knx\\u0027 contains invalid characters. Each segment of the ID must match the pattern [\\\\w-]*.\"}}}"
```

## Additional information

It seems that the UI needs to split the string at every comma and do multiple PUT requests for creating the links.

**Comment by kaikreuzer** [comment]
FTR, I have created https://github.com/openhab/openhab-core/pull/2179 to make the REST API return a 400 instead of a 500 in future.

**Comment by cweitkamp** [comment]
https://github.com/openhab/openhab-core/pull/2179 has been merged.

**Comment by ghys** [comment]
@kaikreuzer I just had a look and it seems that this syntax as documented in https://www.openhab.org/docs/configuration/items.html#multi-binding-channel-linkage works correctly:
```
Dimmer Light_Ceiling "Deckenlicht" <light> { channel="hue:0220:1:20:brightness" [ profile="abc" ], channel="knx:device:ip:knx_1_1_1:3" [ profile="def" ] }
```
with the added benefit that it allows to set the individual links' config, like different profiles.

I don't seem to find any documented example of the comma-separated channels in a single string value, and was unaware it was supported. Is it still relevant with the consideration above (unable to set a different configuration per link). If so how should an eventual configuration reflected in the (multiple) links in that case, like `{channel="channel1,channel2" [ config="value" ]}` - it turns out to be less trivial than expected!

**Comment by kaikreuzer** [comment]
The comma separated syntax was the original one and has been always supported already - and it is imho also much nicer in most cases as it feels slightly weird to have the same key occurring multiple times. Would hence be nice to see it supported in the import as well.

> If so how should an eventual configuration reflected in the (multiple) links in that case

The configuration should be applied to all channels in the list. That's how it is processed in the `GenericItemChannelLinkProvider`, which parses this part of the DSL files.

**Comment by florian-h05** [comment]
@jimtng By using the /rest/file-format endpoint we can get rid of the parser in the UI and fix this issue, correct?

**Comment by lolodomo** [comment]
> [@jimtng](https://github.com/jimtng) By using the /rest/file-format endpoint we can get rid of the parser in the UI and fix this issue, correct?

Yes, that should be do doable and was in the TODO list when the API was developed.

Here is the JSON that will be generated by the new API:
```
{
  "items": [
    {
      "channelLinks": [
        {
          "channelUID": "hue:0220:1:20:brightness"
        },
        {
          "channelUID": "knx:device:ip:knx_1_1_1:3"
        }
      ],
      "type": "Dimmer",
      "name": "Light_Ceiling",
      "label": "Deckenlicht",
      "category": "light",
      "tags": [],
      "groupNames": []
    }
  ]
}
```

PS: if I generate the DSL format from that JSON, the resulting syntax is correct but different from the original because I have not handled this condensed syntax option :
```
Dimmer Light_Ceiling "Deckenlicht" <light> { channel="hue:0220:1:20:brightness", channel="knx:device:ip:knx_1_1_1:3" }
```
Could be changed when there is no configuration paramters, the current code is here:
https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.model.item/src/org/openhab/core/model/item/internal/fileconverter/DslItemFileConverter.java#

**Comment by florian-h05** [comment]
> Yes, that should be do doable and was in the TODO list when the API was developed.

Great, then we should use this.
Only question is how we handle the live-update, as your input is currently passed in real time and displayed in a table.

> PS: if I generate the DSL format from that JSON, the resulting syntax is correct but different from the original

I don’t think it’s a problem.

**Comment by lolodomo** [comment]
> Only question is how we handle the live-update, as your input is currently passed in real time and displayed in a table.

Not sure to understand, what's the difference with current import?
Why are you talking about live-update while the feature is just a punctual import ?
If during import, the item already exists in the registry, I assume it should be override.

**Comment by jsjames** [comment]
@florian-h05 @jimtng @kaikreuzer Just checking the status, now that we have the /rest/file-format implemented in the UI, can this be closed?

**Comment by lolodomo** [comment]
No it has first to be refactored in Main UI to use the new API.
